CI(azure-pipelines): Use static vcpkg environment for Linux build

davidebeatrici opened the pull request:
This will allow us to build static server releases with up-to-date dependencies.

Previously, they were built on an old CentOS machine using Ermine.

The reason why we were not able to do this earlier is microsoft/vcpkg#11727.

@kafeg provided a working workaround.

Krzmbrzl commented:
```
CMake Error at /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168/installed/x64-linux/share/cmake/Qt5DBus/Qt5DBusConfig.cmake:95 (message):
  Library not found: dbus-1
Call Stack (most recent call first):
  /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168/installed/x64-linux/share/cmake/Qt5DBus/Qt5DBusConfig.cmake:277 (_qt5_DBus_process_prl_file)
  /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168/scripts/buildsystems/vcpkg.cmake:461 (_find_package)
  /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168/installed/x64-linux/share/cmake/Qt5Network/Qt5Network_QConnmanEnginePlugin.cmake:8 (find_package)
  /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168/installed/x64-linux/share/cmake/Qt5Network/Qt5NetworkConfig.cmake:433 (include)
  /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168/scripts/buildsystems/vcpkg.cmake:461 (_find_package)
  /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168/installed/x64-linux/share/cmake/Qt5/Qt5Config.cmake:28 (find_package)
  /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168/scripts/buildsystems/vcpkg.cmake:461 (_find_package)
  cmake/pkg-utils.cmake:70 (find_package)
  src/CMakeLists.txt:17 (find_pkg)
```

Krzmbrzl commented:
2 things
1) Is 9761f20a1760c9b1d4312387d5903c878b28bc0b essential for this PR to work? If not I'd say it shouldn't be part of this PR. After all #4427 is dealing with this issue
2) I think our Linux agent runs out of space when unpacking the Linux environment. According to what I have found online the agent has 10 GB of disk space and given that the compressed version of the environment already has 4 GB I think the uncompressed version could get dangerously close to that limit. What is the size of the extracted env?

davidebeatrici commented:
1. `libxcb-xinput-dev` is not available on Ubuntu 16.04.
2. 20+ GB.

Krzmbrzl commented:
Well then according to https://developercommunity.visualstudio.com/content/problem/425838/windows-azure-pipeline-jobs-are-running-out-of-dis.html there are only 10GB available and thus the CI will never succeed...
The only solution to that would be to have a self-hosted machine somehow that has the necessary capacity

davidebeatrici commented:
This doesn't explain why before switching to Ubuntu 18.04 it successfully extracted the archive though.

Krzmbrzl commented:
Indeed :thinking:

Krzmbrzl commented:
Using `axel` to download the environment:
```
Downloaded 4.1 Gigabyte in 4:44 minute(s). (15080.99 KB/s)
```
vs. CI timing out after an hour and only having downloaded 89% using `wget`

I'd say that's a good improvement here :D

Krzmbrzl commented:
Now it's official though - the CI fails with `System.IO.IOException: No space left on device`

We won't be able to use the environment in it current form on this CI server.

davidebeatrici commented:
I wonder why `wget` is so slow... maybe there's a speed limit for single TCP connections on Azure?

davidebeatrici commented:
I removed the entire `debug` directory, resolving the issue.

However:
```
CMake Error at /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168-no-debug/installed/x64-linux/share/protobuf/protobuf-targets.cmake:97 (message):
  The imported target "protobuf::libprotobuf-lite" references the file

     "/opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168-no-debug/installed/x64-linux/debug/lib/libprotobuf-lited.a"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168-no-debug/installed/x64-linux/share/protobuf/protobuf-targets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168-no-debug/installed/x64-linux/share/protobuf/protobuf-config.cmake:8 (include)
  /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168-no-debug/scripts/buildsystems/vcpkg.cmake:461 (_find_package)
  cmake/pkg-utils.cmake:104 (find_package)
  src/CMakeLists.txt:32 (find_pkg)
```

I'm going to restore the libraries mentioned in the file, but this should probably be reported upstream.

davidebeatrici commented:
```
CMake Error at /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168-no-debug/installed/x64-linux/share/libsndfile/LibSndFileTargets.cmake:78 (message):
  The imported target "sndfile-static" references the file

     "/opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168-no-debug/installed/x64-linux/debug/lib/libsndfile.a"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168-no-debug/installed/x64-linux/share/libsndfile/LibSndFileTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168-no-debug/installed/x64-linux/share/libsndfile/LibSndFileConfig.cmake:27 (include)
  /opt/hostedtoolcache/MumbleBuild/linux-static-1.4.x-2020-08-24-f65cd5d-1168-no-debug/scripts/buildsystems/vcpkg.cmake:461 (_find_package)
  cmake/pkg-utils.cmake:104 (find_package)
  src/mumble/CMakeLists.txt:319 (find_pkg)
```

Krzmbrzl commented:
The static client emits this error
```
Fontconfig error: Cannot load default config file
```
and then nothing more happens. It doesn't exit/crash though either :shrug: 

The static server seems to be working fine though.

Krzmbrzl commented:
Here we go again:

> ##[error]Unhandled exception. System.IO.IOException: No space left on device

Krzmbrzl commented:
Closing this as abandoned
